pkg/handler/data_recorder_kinesis.go

```diff
@@ -0,0 +1,87 @@
+package handler
+
+import (
+	"github.com/checkr/flagr/pkg/config"
+	"github.com/checkr/flagr/pkg/util"
+	"github.com/checkr/flagr/swagger_gen/models"
+
+	"github.com/a8m/kinesis-producer"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/sirupsen/logrus"
+)
+
+type kinesisRecorder struct {
+	enabled  bool
+	producer *producer.Producer
+}
+
+// NewKinesisRecorder creates a new Kinesis recorder
+var NewKinesisRecorder = func() DataRecorder {
+	client := kinesis.New(session.New(aws.NewConfig()))
+
+	p := producer.New(&producer.Config{
+		StreamName:          config.Config.RecorderKinesisStreamName,
+		Client:              client,
+		BacklogCount:        config.Config.RecorderKinesisBacklogCount,
+		MaxConnections:      config.Config.RecorderKinesisMaxConnections,
+		FlushInterval:       config.Config.RecorderKinesisFlushInterval,
+		BatchSize:           config.Config.RecorderKinesisBatchSize,
+		BatchCount:          config.Config.RecorderKinesisBatchCount,
+		AggregateBatchCount: config.Config.RecorderKinesisAggregateBatchCount,
+		AggregateBatchSize:  config.Config.RecorderKinesisAggregateBatchSize,
+		Verbose:             config.Config.RecorderKinesisVerbose,
+		Logger:              logrus.WithField("producer", "kinesis"),
+	})
+
+	p.Start()
+
+	go func() {
+		for err := range p.NotifyFailures() {
+			logrus.WithField("kinesis_error", err).Error("error pushing")
+		}
+	}()
+
+	return &kinesisRecorder{
+		producer: p,
+		enabled:  config.Config.RecorderEnabled,
+	}
+}
+
+func (k *kinesisRecorder) AsyncRecord(r *models.EvalResult) {
+	if !k.enabled {
+		return
+	}
+
+	kr := &kinesisEvalResult{
+		EvalResult: r,
+	}
+
+	payload, err := kr.Payload()
+	if err != nil {
+		logrus.WithField("kinesis_error", err).Error("error marshaling")
+	}
+
+	err = k.producer.Put(payload, kr.Key())
+	if err != nil {
+		logrus.WithField("kinesis_error", err).Error("error pushing")
+	}
+}
+
+type kinesisEvalResult struct {
+	*models.EvalResult
+}
+
+// Unmalshalls the context
```

Comment:
nit, comment looks wrong. Do you mean `Payload marshals the EvalResult`?

Reply:
👍